lib/config/presets/index.ts

```diff
@@ -44,6 +44,9 @@ const gitPresetRegex = regEx(
   /^(?<repo>~?[\w\-. /]+)(?::(?<presetName>[\w\-.+/]+))?(?:#(?<tag>[\w\-./]+?))?$/
 );
 
+// shallow mode preset to ignore regex
+const whitesourcePresetRegex = regEx(/^.*>(whitesource)\//);
```

Comment:
```suggestion
const whitesourcePresetRegex = regEx(/^.*>whitesource\/|^(github|gitlab|gitea|local)>/);
```
You could simplify `isPresetForShallowConfig` but the above regex, as you already using `regex.test`